src/test/java/org/wso2/securevault/secret/SecretManagerTest.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.securevault.secret;
+
+import org.powermock.reflect.Whitebox;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Map;
+import java.util.Properties;
+
+public class SecretManagerTest {
+
+    private SecretManager secretManager;
+
+    @BeforeClass
+    public void setUp() {
+
+        secretManager = SecretManager.getInstance();
+    }
+
+    @Test(description = "Test case for filterConfigurations() method. ")
+    public void testFilterConfigurations() throws Exception {
+
+        Properties actual = Whitebox.invokeMethod(secretManager, "filterConfigurations", "secretProviders",
+                getConfigProperties());
+        Assert.assertEquals(12, actual.size());
+    }
+
+    @Test(description = "Test case for addStringToArray() method. ")
```

Comment:
```suggestion
    @Test(description = "Test case for addStringToArray() method.")
```

Check other places as well

Reply:
corrected